dotcom-rendering/src/web/browser/islands/emotion.ts

```diff
@@ -0,0 +1,23 @@
+import type { EmotionCache } from '@emotion/cache';
+import createCache from '@emotion/cache';
+
+/**
+ * Emotion Cache
+ *
+ * Emotion uses a cache to avoid having to re-calculate the same CSS multiple times,
+ * We want to re-use this cache between islands to avoid duplicate work or
+ * conflicts between caches.
+ *
+ * We need to keep this cache available for any islands added to the page after the
+ * initial page load (e.g with liveblogs), so we store it on window.guardian
+ */
+export const getEmotionCache = (): EmotionCache => {
+	if (window.guardian.emotionCache === undefined) {
+		const key = 'dcr';
+		const emotionCache = createCache({ key });
+		window.guardian.emotionCache = emotionCache;
+		return emotionCache;
+	} else {
+		return window.guardian.emotionCache;
+	}
```

Comment:
If you like modern syntax, you could use conditional assignment with `|=`.

This is what the [CMP](https://github.com/guardian/consent-management-platform/blob/843f57e3c40f51a29a6d1a5bd8766a03b9c3852a/src/index.ts#L88-L100) does, for example.